packages/Adapters/STKMesh/example/mesh_deformation.cpp

```diff
@@ -0,0 +1,396 @@
+#include <Teuchos_OpaqueWrapper.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
+#include <Teuchos_TimeMonitor.hpp>
+#include <Teuchos_TypeTraits.hpp>
+#include <Teuchos_VerboseObject.hpp>
+
+#include <Tpetra_MultiVector.hpp>
+
+#include <Intrepid_FieldContainer.hpp>
+
+#include <stk_util/parallel/Parallel.hpp>
+
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/CoordinateSystems.hpp>
+#include <stk_mesh/base/Field.hpp>
+#include <stk_mesh/base/FieldBase.hpp>
+#include <stk_mesh/base/GetEntities.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+#include <stk_mesh/base/Selector.hpp>
+#include <stk_mesh/base/Types.hpp>
+#include <stk_topology/topology.hpp>
+
+#include <stk_io/IossBridge.hpp>
+#include <stk_io/StkMeshIoBroker.hpp>
+
+#include <Ioss_SubSystem.h>
+#include <init/Ionit_Initializer.h>
+
+//---------------------------------------------------------------------------//
+// Set displacement field data on the boundary.
+void setBoundaryDisplacement( stk::mesh::BulkData &bulk_data,
+                              stk::mesh::Selector &fixed_selector,
+                              stk::mesh::Selector &moving_selector,
+                              stk::mesh::Selector &interior_selector,
+                              Teuchos::Array<double> &coord_field,
+                              Teuchos::Array<double> &displace_field, double ux,
+                              double uy, int &num_boundary, int &num_interior )
+{
+    // Get the fixed boundary nodes.
+    stk::mesh::BucketVector fixed_part_buckets =
+        fixed_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> fixed_part_nodes;
+    stk::mesh::get_selected_entities( fixed_selector, fixed_part_buckets,
+                                      fixed_part_nodes );
+    int fixed_num_part_nodes = fixed_part_nodes.size();
+
+    // Get the moving boundary nodes.
+    stk::mesh::BucketVector moving_part_buckets =
+        moving_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> moving_part_nodes;
+    stk::mesh::get_selected_entities( moving_selector, moving_part_buckets,
+                                      moving_part_nodes );
+    int moving_num_part_nodes = moving_part_nodes.size();
+
+    // Get the number of boundary nodes.
+    num_boundary = fixed_num_part_nodes + moving_num_part_nodes;
+
+    // Get the interior nodes.
+    stk::mesh::BucketVector interior_part_buckets =
+        interior_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> interior_part_nodes;
+    stk::mesh::get_selected_entities( interior_selector, interior_part_buckets,
+                                      interior_part_nodes );
+    num_interior = interior_part_nodes.size();
+
+    // Allocate
+    int total_num_nodes = num_boundary + num_interior;
+    coord_field.resize( 3 * total_num_nodes );
+    displace_field.resize( 3 * total_num_nodes );
```

Comment:
This should be a comment in the example. The problem is 2D but you are working as if `dim = 3`, since it is an example you need to document why you are doing it.